crates/agent/src/main.rs

```diff
@@ -47,9 +49,13 @@ async fn main() -> agent::Result<()> {
         }
     };
 
-    let agent_service = AgentService::new(db, config.clone());
+    let agent_service = AgentService::new(db.clone(), config.clone());
 
-    let app = create_router(&config, agent_service);
+    // Initialize cache service
+    let cache_repo = CacheRepository::new(db);
+    let cache_service = CacheService::new(cache_repo, None);
+
+    let app = create_router(&config, agent_service, cache_service);
```

Comment:
Pass optional cache service instead of `None` boilerplate
```suggestion
  let cache_repo = CacheRepository::new(db);
  let cache_service = CacheService::new(cache_repo);
  let app = create_router(&config, agent_service, Some(cache_service));

```

Reply:
non-optional